### PR TITLE
Use same addons script for init.d and systemd.

### PR DESCRIPTION
**Pull request by erictune**

### CODE REVIEW

**Comment by erictune** [comment]
Testing with kube-up.sh in progress.

**Comment by zmerlynn** [comment]
cc @derekwaynecarr, you may want to test this too

**Comment by derekwaynecarr** [comment]
I just merged https://github.com/GoogleCloudPlatform/kubernetes/pull/6382 which fixed vagrant after yesterday's changes with etcd in pod.  If it's not urgent, I cannot test this until early next week (out tomorrow).

File: cluster/saltbase/salt/kube-addons/kube-addons.sh [hunk 1]
```diff
@@ -19,22 +19,31 @@
...
-    if ${KUBECTL} --server="127.0.0.1:8080" create --validate=true -f ${obj}; then
-      return
-    fi
-    echo "++ ${obj} failed, attempt ${try} (sleeping 5) ++"
-    sleep 5
+# $1 addon to start.
+# $2 count of tries to start the addon.
+# $3 delay in seconds between two consecutive tries
+function start_addon() {
+  addon=$1;
```
Comment (zmerlynn) [inline cluster/saltbase/salt/kube-addons/kube-addons.sh]: Make these locals in this PR and save some time in the other? :)

**Comment by zmerlynn** [comment]
LGTM except minor nit (and obviously testing)

**Comment by erictune** [comment]
Tested by running kube-up and verifying all addons started.

**Comment by erictune** [comment]
Not going to fix nit since I already tested it and don't want to wait for another round of testing.

**Comment by zmerlynn** [comment]
K. :)

**Pull request merged**